eq-author/src/components/PositionModal/index.js

```diff
@@ -32,64 +32,123 @@ const Trigger = styled.button.attrs({ type: "button" })`
   }
 `;
 
+const Indent = styled(Option)`
+  margin-left: ${({ indent }) => (indent ? 1 : 0)}em;
+`;
+
 const PositionModal = ({ options, onMove, selected }) => {
+  const positionButtonId = uniqueId("PositionModal");
+  const [isOpen, setIsOpen] = useState(false);
   const previousPosition = useMemo(
     () => options.findIndex(({ id }) => id === selected.id),
     [options, selected]
   );
-  const [isOpen, setIsOpen] = useState(false);
-  const [selectedPosition, setSelectedPosition] = useState(previousPosition);
-  const [positionButtonId] = useState(uniqueId("PositionModal"));
+
+  const [{ position, item, isSame }, setOption] = useState({
+    position: previousPosition,
+    item: options[previousPosition],
+  });
 
   const orderedOptions = options.filter(({ id }) => id !== selected.id);
-  orderedOptions.splice(selectedPosition, 0, selected);
+  selected.parentEnabled = item?.parentEnabled;
+  orderedOptions.splice(position, 0, selected);
 
   const handleClose = () => {
     setIsOpen(false);
-    setSelectedPosition(previousPosition);
+    setOption({
+      position: previousPosition,
+      item: orderedOptions[previousPosition],
+    });
   };
 
-  const handleOpen = () => setIsOpen(true);
-
-  const handleChange = ({ value }) => setSelectedPosition(parseInt(value, 10));
+  const handleChange = ({ value }) =>
+    setOption(({ item: previousItem }) => {
+      let count =
```

Comment:
```suggestion
      const count =
```

I think? I can't see it being modified anywhere else, but I might be wrong.